.github/workflows/main.yaml

```diff
@@ -0,0 +1,97 @@
+name: CI/CD
+
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on push or pull request events but only for the main branch
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: '*'
+
+env:
+  CONTAINER_REGISTRY: ghcr.io
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains two jobs named "build" and "container"
+  build:
+    name: Compile and test
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Setup node
+        uses: actions/setup-node@v3
+        with:
+          node-version-file: '.node-version'
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Generate code
+        run: npm run generate
+
+      - name: Compile
+        run: npm run compile
+
+      - name: Lint code
+        run: npm run lint
+
+      - name: Execute tests
+        run: npm test
+
+  build-container:
+    name: Build container image
+    needs: build
+    runs-on: ubuntu-latest
+    env:
+      image_tag: ${{ github.repository }}:${{ github.sha }}
+
+    steps:
+      - name: Build container image
+        uses: docker/build-push-action@v3
+        with:
+          tags: ${{ env.image_tag }}
+
+      - name: Run Trivy vulnerability scanner
+        uses: aquasecurity/trivy-action@0.7.1
+        with:
+          image-ref: ${{ env.image_tag }}
+          format: 'template'
+          template: '@/contrib/sarif.tpl'
+          output: 'trivy-results.sarif'
+          severity: 'CRITICAL,HIGH'
+          security-checks: 'vuln,secret,config'
+
+      - name: Upload Trivy scan results to GitHub Security tab
```

Comment:
My only doubt here is that we have to do the repo checkout in this job as well.
If you look at [this](https://github.com/pagopa/pn-local-emulator/actions/runs/3188556831/jobs/5201368873) log and take a look at the log of the "Upload Trivy scan [...]", you can see that it fails giving the error message "fatal: not a git repository (or any of the parent directories): .git"

Reply:
Good catch !🔥! The issue is tracked [here](https://github.com/github/codeql-action/issues/944). I'm going to add the checkout step.

Reply:
Update: I'm using an old version of this action. I'm going to update the version of the action and see if it is fixed